priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2017 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.priam;
+
+import com.amazonaws.services.simpledb.AmazonSimpleDB;
+import com.amazonaws.services.simpledb.AmazonSimpleDBClient;
+import com.amazonaws.services.simpledb.model.Attribute;
+import com.amazonaws.services.simpledb.model.Item;
+import com.amazonaws.services.simpledb.model.SelectRequest;
+import com.amazonaws.services.simpledb.model.SelectResult;
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagement;
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClient;
+import com.amazonaws.services.simplesystemsmanagement.model.*;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Loads config data from SSM Paramter Store.  {@link #intialize(String, String)} will query Parameter store for any keys with prefix "PriamProperties-{apppId}-"
+ * for any potential configurations.  The prefix is set up to support multiple different clusters; this is done by using
+ * amazon's auto scaling groups (ASG) to infer the appId in the same manner as SimpleDBConfigSource.
+ */
+public final class SSMParameterStoreConfigSource extends AbstractConfigSource {
+    private static final Logger logger = LoggerFactory.getLogger(SSMParameterStoreConfigSource.class.getName());
+
+    private static final String PRIAM_PREFIX = "PriamProperties";
+
+    private final Map<String, String> data = Maps.newConcurrentMap();
+    private final ICredential provider;
+
+    @Inject
+    public SSMParameterStoreConfigSource(final ICredential provider) {
+        this.provider = provider;
+    }
+
+    @Override
+    public void intialize(final String appid, final String region) {
```

Comment:
initialize* (need to also correct in the comment)

Reply:
This is the interface's name for it. Gonna keep the spelling mistake to keep closer to base. Thanks though.

Reply:
Oh dear. OK, makes sense.